### PR TITLE
convert the Promote page from ERB to partials

### PR DESCRIPTION
**Pull request by Hamms**

### CODE REVIEW

File: pegasus/sites.v3/hourofcode.com/views/promote/sample_emails.erb [hunk 1]
```diff
@@ -0,0 +1 @@
+<%= resolve_url('/promote/resources#sample-emails') %>
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/views/promote/sample_emails.erb:1]: Does the `#sample-emails` part do anything in this call to `resolve_url`?

File: pegasus/sites.v3/hourofcode.com/public/promote/index.md.partial [hunk 2]
```diff
@@ -0,0 +1,28 @@
+---
+title: Spread the word
+layout: wide
+nav: promote_nav
+---
+
+{{ signup_button }}
```
Comment (breville) [inline pegasus/sites.v3/hourofcode.com/public/promote/index.md.partial:7]: Oh, is `{{ }}` the new syntax for including partials in a `.partial`?

File: pegasus/sites.v3/hourofcode.com/views/promote/sample_emails.erb (hunk 1 above)
Comment (breville) [reply pegasus/sites.v3/hourofcode.com/views/promote/sample_emails.erb:1]: Oh nevermind, I get it now.

File: pegasus/sites.v3/hourofcode.com/public/promote/index.md.partial (hunk 2 above)
Comment (Hamms) [reply pegasus/sites.v3/hourofcode.com/public/promote/index.md.partial:7]: Oh, yep! Sorry I wasn't clear about that; this is a follow-up to https://github.com/code-dot-org/code-dot-org/pull/28118

**Review by breville — APPROVED** [review]

**Pull request merged**